### PR TITLE
Update docs for AttributeFilter components

### PR DESCRIPTION
**Pull request by kandl**
- make AttributeFilter components documentation up-to-date, add details about new features, update screenshots
- update page that describes options for custom attribute filter implementation

JIRA: RAIL-4258

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1156/ : SUCCESS in 1m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/771/ : SUCCESS in 8s

File: website/sidebars.json [hunk 1]
```diff
@@ -63,7 +63,6 @@
         ],
         "Properties": [
             "chart_config",
-            "attribute_element_option",
```
Comment (no23reason) [inline website/sidebars.json:66]: Hide this page in siteConfig as well please.

File: docs/30_tips__filter_visual_components.md [hunk 2]
```diff
@@ -344,7 +343,7 @@ const style = { height: 300 };
     />
 </div>
 ```
-
+//
```
Comment (no23reason) [inline docs/30_tips__filter_visual_components.md:346]: Should this be removed?

File: docs/30_tips__create_custom_attribute_filter.md [hunk 3]
```diff
@@ -5,17 +5,33 @@ sidebar_label: Create a Custom Attribute Filter
 copyright: (C) 2007-2020 GoodData Corporation
 ---
 
-The **Attribute Elements component** is a low-level counterpart to the [Attribute Filter component](10_vis__attribute_filter_component.md).
+To implement a custom attribute filter, you have few options:
```
Comment (no23reason) [inline docs/30_tips__create_custom_attribute_filter.md:8]: Please also describe the non-React way using the Handler classes directly.

File: docs/30_tips__create_custom_attribute_filter.md [hunk 4]
```diff
@@ -5,17 +5,33 @@ sidebar_label: Create a Custom Attribute Filter
...
+## useAttributeFilterController hook
+This is the best option if you need to implement fully custom UI for the attribute filter. This option requires a bit more coding, but you have a full control over the UI.
 
-### Example
+It has identical convenient API as [AttributeFilter](attribute_filter_component) component - same input props and same output props that are available in the [internal context](attribute_filter_component#accessing-internal-attributefilter-context) of the AttributeFilter component.
 
-For the latest up-to-date examples, see the [live examples](https://github.com/gooddata/gooddata-ui-sdk/blob/master/examples/sdk-examples/src/examples/attributeFilter/AttributeElementsExample.tsx) in the GoodData.UI repository.
+It works out of the box with other UI.SDK things - [BackendProvider](https://sdk.gooddata.com/gooddata-ui-apidocs/docs/sdk-ui.backendprovider.html), [WorkspaceProvider](https://sdk.gooddata.com/gooddata-ui-apidocs/docs/sdk-ui.workspaceprovider.html) and [visualization definition placeholders](placeholders).
+
+See the [live code example]().
```
Comment (no23reason) [inline docs/30_tips__create_custom_attribute_filter.md:22]: Missing link?

File: docs/30_tips__create_custom_attribute_filter.md [hunk 5]
```diff
@@ -5,17 +5,33 @@ sidebar_label: Create a Custom Attribute Filter
...
+
+## IAttributeFilterHandler API
+This is the best option if you can't or don't want to use React. It requires the most coding of all the mentioned options, but it also offers the most control for you custom attribute filter implementation.
+- It's framework agnostic JavaScript class, so you can integrate it with any UI framework, or even use it in a headless environment.
+- It offers more low level APIs, such as firing custom attribute elements requests and subscribtion to various events.
+- It has support for both multi or single selection.
+- It still does a lot of the thing for you out of the box - eg. loading of the attribute metadata and elements selection during the initialization, and much more.
+
+[Explore its interface]() or see the [live code example]().
```
Comment (no23reason) [inline docs/30_tips__create_custom_attribute_filter.md:37]: Missing links?

File: website/sidebars.json (hunk 1 above)
Comment (kandl) [reply website/sidebars.json:66]: I think that in siteConfig, it's already included see https://github.com/gooddata/gooddata-ui/blob/master/website/siteConfig.js#L75-L77

File: docs/30_tips__create_custom_attribute_filter.md (hunk 3 above)
Comment (kandl) [reply docs/30_tips__create_custom_attribute_filter.md:8]: It's mentioned from LOC 30 and below, or I'm not sure what do you mean?

File: docs/30_tips__filter_visual_components.md (hunk 2 above)
Comment (kandl) [reply docs/30_tips__filter_visual_components.md:346]: Removed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1157/ : SUCCESS in 1m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/772/ : SUCCESS in 7s

File: website/sidebars.json (hunk 1 above)
Comment (no23reason) [reply website/sidebars.json:66]: Oh, my bad :)

File: docs/30_tips__create_custom_attribute_filter.md (hunk 3 above)
Comment (no23reason) [reply docs/30_tips__create_custom_attribute_filter.md:8]: I am blind 🤦‍♂️

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1158/ : SUCCESS in 1m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/773/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1160/ : SUCCESS in 1m 07s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-oss-check/775/ : SUCCESS in 15s

**Review by hackerstanislav — APPROVED** [review]

**Review by ivanmjartan — APPROVED** [review]

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-gate-PR556

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-ui-documentation-validation/1161/ : SUCCESS in 1m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-fossa-scanning-zuul-el7/257/ : FAILURE in 2m 16s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sonarqube-scan-zuul-gate/152/ : SUCCESS in 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-merge-compliance-check/347/ : SUCCESS in 24s

**Pull request merged**